internal/radix/doc.go

```diff
@@ -0,0 +1,4 @@
+// This packages contents were originally copied from github.com/armon/go-radix.
+// After the intial copy all the data structures were made to use Go 1.18 generics
+// instead of relying on the use of interface{} or the any type.
+package radix
```

Comment:
Is the plan to contribute this upstream? Perhaps as a `v2` of the module? Or keep it in-tree, if you'll excuse the pun 🌴 😄

Reply:
I actually have a branch with all these changes in it. I kind of didn't want to open that can of worms yet. I could see publishing the changes to github.com/hashicorp/go-radix (out of the armon namespace).

Reply:
I have PR'ed it upstream. We will see if Armon wants to merge it: https://github.com/armon/go-radix/pull/23